### PR TITLE
use HostAndPort in domain parsing

### LINKED ISSUE #2072: 2.2.16 On http://localhost:8888 server all pictures paths are without port
On http://localhost:8888 server pictures paths are without port.
http://localhost/uploads/favicon.png instead of http://localhost:8888/uploads/favicon.png

vanilla 2.2.16

### PR DESCRIPTION
**Pull request by imnotjames**
Since domain includes the scheme it's not that far fetched for it to have the port as well.  This fixes assets that are loaded with domain pointing at the wrong URL.

Fixes #2072

### CODE REVIEW

File: library/core/class.request.php [hunk 1]
```diff
@@ -341,8 +341,14 @@ protected function _LoadEnvironment() {
       $this->_EnvironmentElement('ConfigWebRoot', Gdn::Config('Garden.WebRoot'));
       $this->_EnvironmentElement('ConfigStripUrls', Gdn::Config('Garden.StripWebRoot', FALSE));
 
-      $this->RequestHost(     isset($_SERVER['HTTP_X_FORWARDED_HOST']) ? val('HTTP_X_FORWARDED_HOST',$_SERVER) : (isset($_SERVER['HTTP_HOST']) ? val('HTTP_HOST',$_SERVER) : val('SERVER_NAME',$_SERVER)));
-      $this->RequestMethod(   isset($_SERVER['REQUEST_METHOD']) ? val('REQUEST_METHOD',$_SERVER) : 'CONSOLE');
+      $Host = isset($_SERVER['HTTP_X_FORWARDED_HOST']) ? val('HTTP_X_FORWARDED_HOST',$_SERVER) : (isset($_SERVER['HTTP_HOST']) ? val('HTTP_HOST',$_SERVER) : val('SERVER_NAME',$_SERVER));
```
Comment (kaecyra) [inline library/core/class.request.php]: I'm not a big fan of this inscrutable nested ternary if-statement. Would you mind expanding it?

**Comment by Malet** [comment]
+1 this fixes an issue I was having when running on `localhost:3000` :+1:

File: library/core/class.request.php (hunk 1 above)
Comment (Malet) [inline library/core/class.request.php]: Also `val()` doesn't actually do anything in this case, better off using `$_SERVER['SERVER_NAME']` etc.

**Comment by kaecyra** [comment]
Bump @imnotjames, this needs cleanup per comment.

**Comment by linc** [comment]
Note to self: This is confirmed clear, besides the syntax issue.

File: library/core/class.request.php (hunk 1 above)
Comment (imnotjames) [inline library/core/class.request.php]: The `val()` function does very little in almost every case where it is used except make the code a little more indirect.

**Comment by imnotjames** [comment]
I wasn't a fan of the ternary either but used it because it was in the original code.  Updated.

**Comment by linc** [comment]
Thanks!

**Pull request merged**